### PR TITLE
Add react-query

### PR DESCRIPTION
**Pull request by aguscha333**
This PR adds the basic react query integration and some example models, services and queries that devs can base their queries on. The example is not functional and because of that was not tested but it was generated based on similar code from an existing project.

### CODE REVIEW

**Review by santiagorobra — APPROVED** [review]
🚀

File: src/network/models/example-models.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  };
+};
+
+export type ExampleResponse = {
+  pet: {
+    id: string;
+    name: string;
+    age?: number;
+  };
+};
```
Comment (juanchoperezj) [inline src/network/models/example-models.ts:22]: I saw a few days ago this library [zod](https://zod.dev/?id=basic-usage). Will depend on the use case but could be useful.

**Review by juanchoperezj — APPROVED** [review]
💯

File: src/network/models/example-models.ts (hunk 1 above)
Comment (aguscha333) [reply src/network/models/example-models.ts:22]: that is for validations, not sure if it helps you define ts models for the endpoints

**Review by marcos09 — APPROVED** [review]

**Pull request merged**